src/transformers/utils/quantization_config.py

```diff
@@ -418,3 +423,51 @@ def post_init(self):
                     f"""dataset needs to be either a list of string or a value in
                     ['wikitext2','c4','c4-new','ptb','ptb-new'], but we found {self.dataset}"""
                 )
+        if self.bits == 4:
+            if self.disable_exllama is not None:
+                logger.warning(
+                    "Using `disable_exllama` is deprecated and will be removed in future version. Use `use_exllama` instead."
+                )
+                self.use_exllama = not self.disable_exllama
+
+            if self.use_exllama_v2:
+                optimum_version = version.parse(importlib.metadata.version("optimum"))
+                autogptq_version = version.parse(importlib.metadata.version("auto_gptq"))
+                if optimum_version <= version.parse("1.13.2") or autogptq_version <= version.parse("0.4.2"):
+                    raise ValueError(
+                        f"You need optimum > 1.13.2 and auto-gptq > 0.4.2 . Make sure to have that version installed - detected version : optimum {optimum_version} and autogptq {autogptq_version}"
+                    )
+                self.use_exllama = False
+                logger.warning("You have activated exllamav2 kernels. Exllama kernels will be disabled.")
+
+            if self.use_exllama:
+                logger.warning(
+                    "You have activated exllama backend. Note that you can get better inference "
+                    "speed using exllamav2 kernel by setting `use_exllama_v2=True`."
+                )
+
+    def to_dict_optimum(self):
```

Comment:
Why not just `to_dict`?

Reply:
Optimum configuration is not the same as transformers one, hence the mapping.

Reply:
Yes, since optimum config is not the same. It also has the `disable_exllama` arg. I will do the same modification as here so that we don't end up having different config. cc @fxmarty 

```py
# This is for the transformers library -> Will be saved in the config.json
config.to_dict()
# This is for the optimum library -> we need to match transformers config with optimum config. 
config.to_dict_optimum()
```